edward/inferences/variational_inference.py

```diff
@@ -53,25 +53,70 @@ def initialize(self, optimizer=None, var_list=None, use_prettytensor=False,
     """
     super(VariationalInference, self).initialize(*args, **kwargs)
 
+    latent_var_list = set()
+    data_var_list = set()
     if var_list is None:
       # Traverse random variable graphs to get default list of variables.
-      var_list = set()
       trainables = tf.trainable_variables()
       for z, qz in six.iteritems(self.latent_vars):
         if isinstance(z, RandomVariable):
-          var_list.update(get_variables(z, collection=trainables))
+          latent_var_list.update(get_variables(z, collection=trainables))
 
-        var_list.update(get_variables(qz, collection=trainables))
+        latent_var_list.update(get_variables(qz, collection=trainables))
 
       for x, qx in six.iteritems(self.data):
         if isinstance(x, RandomVariable) and \
                 not isinstance(qx, RandomVariable):
-          var_list.update(get_variables(x, collection=trainables))
+          data_var_list.update(get_variables(x, collection=trainables))
 
-      var_list = list(var_list)
+      var_list = list(data_var_list | latent_var_list)
 
     self.loss, grads_and_vars = self.build_loss_and_gradients(var_list)
 
+    if self.logging:
+      summary_key = 'summaries_' + str(id(self))
+      tf.summary.scalar("loss", self.loss, collections=[summary_key])
+      with tf.name_scope('variational'):
+        for grad, var in grads_and_vars:
+          if var in latent_var_list:
+            tf.summary.histogram("parameter_" +
+                                 var.name.replace(':', '_'),
+                                 var, collections=[summary_key])
+            tf.summary.histogram("gradient_" +
+                                 grad.name.replace(':', '_'),
+                                 grad, collections=[summary_key])
+            tf.summary.scalar("gradient_norm_" +
+                              grad.name.replace(':', '_'),
+                              tf.norm(grad), collections=[summary_key])
+      # replace : with _ because tf does not allow : in var names in summaries
```

Comment:
@akshaykhatri639 Really looking forward to this feature!
I tried it using MAP inference, but the naming turns out weird.
     
![tensorboard](https://cloud.githubusercontent.com/assets/1409833/25719871/e27bac12-30d8-11e7-94b9-72dd5516c057.jpg)

I would replace `grad.name.replace(':', '_')` with `var.name.replace(':', '_')` everywhere, because the gradients are named after the operations. But then by prefacing it with `gradient_` it will still be clear from the name that its a gradient.

I have been using `tf.name_scope('model')` for the actual model.
![tb_model](https://cloud.githubusercontent.com/assets/1409833/25720091/a8109c58-30d9-11e7-834c-3c7333ebbc25.jpg)
 
Maybe rename to `tf.name_scope('training')` for logging? But that's minor. I can just rename the scope for my model.